code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm

```diff
@@ -272,6 +272,13 @@ datum/reagent/water/reaction_obj(var/obj/O, var/volume)
 			cube.Expand()
 	return
 
+datum/reagent/water/reaction_obj(var/obj/O, var/volume)
+	src = null
+	if(istype(O,/obj/item/dehy_carp))
+		var/obj/item/dehy_carp/dehy = O
+		dehy.Swell() // Makes a carp
+		return
```

Comment:
As I'm only just breaking my way into understanding how to code in this, I'm not entirely sure how. To be honest that is effectively the same code as for the monkey cubes minus a check for whether or not the cube is still wrapped up.